components/TaskList.tsx

```diff
@@ -21,12 +21,32 @@ export default function TaskList({ cat }) {
     }
 
     fetchTasks();
-  }, []);
+  }, [db]);
 
   function toggle() {
     setIsDisplayed(!isDisplayed);
   }
 
+  function onDelete(id) {
+    setTasks((curState) => {
+      const taskItemIndex = curState.findIndex((item) => item.id === id);
+
+      if (taskItemIndex != -1) curState.splice(taskItemIndex, 1);
+
+      return curState;
+    });
+
+    function deleteTask() {
+      db.runAsync("DELETE FROM Tasks WHERE id = ?", [id])
+        .then((result) => {
+          console.log(result);
+        })
+        .catch((err) => console.log("Error:", err));
+    }
+
+    deleteTask();
```

Comment:
There is no point of create a function within the same function and invoking it right after definition. Inline the function's code into outer scope. 

Since it's an async task you might as well use `await` and mark the function as `async`